subsys/bluetooth/mesh/light_hsl_cli.c

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/mesh/light_hsl_cli.h>
+#include "model_utils.h"
+#include "light_hsl_internal.h"
+
+static void status_decode(struct bt_mesh_light_hsl_cli *cli,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf, uint32_t opcode,
+			  struct bt_mesh_light_hsl_status *status)
+{
+	if (buf->len != BT_MESH_LIGHT_HSL_MSG_MINLEN_STATUS &&
+	    buf->len != BT_MESH_LIGHT_HSL_MSG_MAXLEN_STATUS) {
+		return;
+	}
+
+	light_hsl_buf_pull(buf, &status->params);
+	status->remaining_time =
+		(buf->len == 1) ?
+			model_transition_decode(net_buf_simple_pull_u8(buf)) :
+			0;
+
+	if (model_ack_match(&cli->ack_ctx, opcode, ctx)) {
+		struct bt_mesh_light_hsl_status *rsp =
+			(struct bt_mesh_light_hsl_status *)
+				cli->ack_ctx.user_data;
+		*rsp = *status;
+		model_ack_rx(&cli->ack_ctx);
+	}
+}
+
+static void hsl_status_handle(struct bt_mesh_model *model,
+				  struct bt_mesh_msg_ctx *ctx,
+				  struct net_buf_simple *buf)
+{
+	struct bt_mesh_light_hsl_cli *cli = model->user_data;
+	struct bt_mesh_light_hsl_status status;
+
+	status_decode(cli, ctx, buf, BT_MESH_LIGHT_HSL_OP_STATUS, &status);
+
+	if (cli->handlers->status) {
```

Comment:
I guess there is a potential bug here. `status` is the local not initialized variable. That might be any. If `status_decode` return context quickly due to length error the client will raise up garbage to app.
We need to check `status_decode` result here and below.